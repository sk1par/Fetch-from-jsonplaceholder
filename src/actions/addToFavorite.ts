import IUser from "../models/user.interface";

const history = (user: IUser) => {
    return {
        type: 'ADD_FAVORITE',
        payload: user
    }
}

export default history;