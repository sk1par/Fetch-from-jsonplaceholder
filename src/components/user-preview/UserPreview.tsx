import { useDispatch, useSelector } from 'react-redux';
import { addToFavorite } from '../../actions';
import { selectUser } from '../../selectors/selectors';

const UserPreview = () => {
  const dispatch = useDispatch();
  const user = useSelector(selectUser);

  const addToFavorites = () => {
    dispatch(addToFavorite(user));
};

  return (
    <div className="card" style={{ width: "25rem", margin: "auto" }} data-auto-id='user-preview-div'>
      <div className="card-body">
        <h5 className="card-title">Name: {user?.name}</h5>
        <h6 className="card-subtitle mb-2 text-muted">Username: {user?.username}</h6>
        <a href={"tel:" + user?.phone} className="card-link">{user?.phone}</a>
        <a href={"mailto:" + user?.email} className="card-link">{user?.email}</a>
      </div>
      <div className='button-div'>
        <button type="button" style={{ marginBottom: '5px' }} className="btn btn-outline-success"
          onClick={addToFavorites} data-auto-id='btn'>Add to Favorites</button>
      </div>
    </div>
  )
}

export default UserPreview
