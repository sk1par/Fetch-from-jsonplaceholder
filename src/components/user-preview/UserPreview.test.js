import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import UserPreview from './UserPreview';
import { useSelector } from 'react-redux';
import  addToFavorite from '../../actions/addToFavorite';


Enzyme.configure({ adapter: new Adapter() });
jest.mock('react-redux', () => ({
    useDispatch: () => (fn) => () => fn(),
    useSelector: jest.fn(() => []),
}));
jest.mock('../../actions/addToFavorite', () => {
    const addToFavorite = jest.fn();
    return addToFavorite;
});

const getInitialObject = () => [{
    email: 'test@abv.bg',
    id: 1,
    name: 'test',
    phone: '123132',
    username: 'skipar',
    website: 'abv.bg'
}];

it('should render User Preview component correctly', () => {
    const wrapper = shallow(<UserPreview />);
    const userPreviewDiv = wrapper.find('[data-auto-id="user-preview-div"]');
    expect(userPreviewDiv).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});

it.only('should trigger add to favorite action correctly after click', () => {
    const value = getInitialObject();
    useSelector.mockReturnValueOnce(value);
    const wrapper = mount(<UserPreview />);
    const item = wrapper.find('[data-auto-id="btn"]');
    item.simulate('click');
    expect(addToFavorite).toHaveBeenCalledTimes(1);
    expect(addToFavorite).toHaveBeenCalledWith(value);
    expect(wrapper).toMatchSnapshot();
});