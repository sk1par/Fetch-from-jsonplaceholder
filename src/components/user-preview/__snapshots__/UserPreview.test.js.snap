// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render User Preview component correctly 1`] = `ShallowWrapper {}`;

exports[`should trigger add to favorite action correctly after click 1`] = `ReactWrapper {}`;
